pkg/services/cleanup/cleanup.go

```diff
@@ -418,3 +425,133 @@ func (srv *CleanUpService) cleanUpTrashAlertRules(ctx context.Context) {
+// cascading deletions on teams to delete existing lbac rules
+func (srv *CleanUpService) cleanupStaleLBACRules(ctx context.Context) {
+	logger := srv.log.FromContext(ctx)
+
+	// Get all datasources
+	allDataSources, err := srv.dataSourceService.GetAllDataSources(ctx, &datasources.GetAllDataSourcesQuery{})
+	if err != nil {
+		logger.Error("Failed to get datasources for LBAC cleanup", "error", err)
+		return
+	}
+
+	var totalCleaned int
+	var totalDataSources int
+
+	for _, ds := range allDataSources {
+		if ds.JsonData == nil {
+			continue
+		}
+
+		// Check if datasource has team LBAC rules
+		teamHTTPHeaders, err := datasources.GetTeamHTTPHeaders(ds.JsonData)
+		if err != nil || teamHTTPHeaders == nil {
+			continue
+		}
+
+		totalDataSources++
+
+		// Extract team UIDs and check if teams still exist
+		cleanedRules, removedCount := srv.getLBACRulesForTeamsStillExisting(ctx, teamHTTPHeaders, ds.OrgID)
+
+		if removedCount > 0 {
+			// Update the datasource with cleaned rules
+			err := srv.updateDataSourceLBACRules(ctx, ds, cleanedRules)
+			if err != nil {
+				logger.Error("Failed to update datasource LBAC rules",
+					"datasource", ds.UID, "error", err)
+			} else {
+				totalCleaned += removedCount
+				logger.Debug("Cleaned stale LBAC rules",
+					"datasource", ds.UID, "removed", removedCount)
+			}
+		}
+	}
+
+	if totalCleaned > 0 {
+		logger.Info("Cleaned up stale team LBAC rules",
+			"datasources_processed", totalDataSources,
+			"total_rules_removed", totalCleaned)
+	}
+}
+
+func (srv *CleanUpService) getLBACRulesForTeamsStillExisting(ctx context.Context, teamHeaders *datasources.TeamHTTPHeaders, orgID int64) (*datasources.TeamHTTPHeaders, int) {
+	logger := srv.log.FromContext(ctx)
+	cleanedHeaders := &datasources.TeamHTTPHeaders{Headers: make(map[string][]datasources.TeamHTTPHeader)}
+	removedCount := 0
+
+	for teamIdentifier, headers := range teamHeaders.Headers {
+		// Determine if this is a UID or ID
+		var teamUID string
+		teamID, err := strconv.ParseInt(teamIdentifier, 10, 64)
+
+		if err != nil {
+			// It's a UID
+			teamUID = teamIdentifier
+		} else {
+			// It's an ID, need to resolve to UID
+			teamByID, err := srv.teamService.GetTeamByID(ctx, &team.GetTeamByIDQuery{
+				OrgID: orgID,
+				ID:    teamID,
+			})
+			if err != nil {
+				logger.Debug("Team ID no longer exists, removing LBAC rules",
+					"teamID", teamIdentifier, "orgID", orgID)
+				removedCount++
+				continue
+			}
+			teamUID = teamByID.UID
+		}
+
+		// Check if team still exists by UID
+		_, err = srv.teamService.GetTeamByID(ctx, &team.GetTeamByIDQuery{
+			OrgID: orgID,
+			UID:   teamUID,
+		})
+
+		if err != nil {
+			logger.Debug("Team UID no longer exists, removing LBAC rules",
+				"teamUID", teamUID, "orgID", orgID)
+			removedCount++
+			continue
+		}
+
+		// Team exists, keep the rules
+		cleanedHeaders.Headers[teamIdentifier] = headers
```

Comment:
When we build lbac rules to send to dsproxy, do we use UID or numeric ID?  Whichever we use, we should save the cleaned up rules with that ID so we don't have to do a lookup later.